File: Java/DeepSpace/src/deepspace/Loot.java
package deepspace;


import java.util.logging.Logger;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author victor
 */
class Loot {
    private int nSupplies, nWeapons, nShields, nHangars, nMedals;

    Loot(int nSupplies, int nWeapons, int nShields, int nHangars, int nMedals) {
        this.nSupplies = nSupplies;
        this.nWeapons = nWeapons;
        this.nShields = nShields;
        this.nHangars = nHangars;
        this.nMedals = nMedals;
    }   

    public int getNSupplies() {
        return nSupplies;
    }

    public int getNWeapons() {
        return nWeapons;
    }

    public int getNShields() {
        return nShields;
    }

    public int getNHangars() {
        return nHangars;
    }

    public int getNMedals() {
        return nMedals;
    }
    
    LootToUI getUIversion() {
        return new LootToUI(this);
    }

    @Override
    public String toString() {
        return "Loot{" + "nSupplies=" + nSupplies + ", nWeapons=" + nWeapons + ", nShields=" + nShields + ", nHangars=" + nHangars + ", nMedals=" + nMedals + '}';
    }
    
    
    
    
    
}

File: Deepspace_ruby/Deepspace/lib/Loot.rb
#encoding:utf-8
module Deepspace
class Loot
  attr_reader :nSupplies, :nWeapons, :nShields, :nHangars, :nMedals 
  
  def initialize(nSupplies, nWeapons, nShields, nHangars,nMedals)
    @nSupplies = nSupplies
    @nWeapons = nWeapons
    @nShields = nShields
    @nHangars = nHangars
    @nMedals = nMedals    
  end
  
   def to_s
    "\nSupplies: #{@nSupplies}, nWeapons: #{@nWeapons} , nShields: #{@nShields}" +
      "nHangars: #{@nHangars}, nMedals: #{@nMedals}"
  end
  
   def getUIversion
     LootToUI.new(self)
   end
   
   def to_s
       return "\nnSupplies = #{@nSupplies} nWeapons = #{@nWeapons} nShields = #{@nShields} nHangars = #{@nHangars} Medals = #{@nMedals}"
  end 
  
end
end